### PR TITLE
create one way messages feature

### PR DESCRIPTION
**Pull request by vikstrous**
I implemented one way messages by making callbacks optional on the client side and using serializers that return Unit as one way serializers, indicating that there can be no response.

This is a better way of doing what I did here: https://github.com/linkedin-sna/norbert/pull/22

### CODE REVIEW

File: network/src/main/scala/com/linkedin/norbert/network/server/MessageExecutorComponent.scala [hunk 1]
```diff
@@ -125,7 +125,7 @@ class ThreadPoolMessageExecutor(clientName: Option[String],
       if(now - queuedAt > requestTimeout) {
         totalNumRejected.incrementAndGet
         log.warn("Request timed out, ignoring! Currently = " + now + ". Queued at = " + queuedAt + ". Timeout = " + requestTimeout)
-        callback(Left(new HeavyLoadException))
+        if(!callback.isEmpty) callback.get(Left(new HeavyLoadException))
```
Comment (jhartman) [inline network/src/main/scala/com/linkedin/norbert/network/server/MessageExecutorComponent.scala]: This does not change the functionality of the code, but you may want to consider the functional style:
callback.foreach(cb => cb(Left(new HeavyLoadException))

**Comment by jhartman** [comment]
I think you should try to add some more unit tests for the newer code. I don't think you've tested the case where you are sending the message one way only.

File: network/src/main/scala/com/linkedin/norbert/network/server/MessageExecutorComponent.scala (hunk 1 above)
Comment (vikstrous) [inline network/src/main/scala/com/linkedin/norbert/network/server/MessageExecutorComponent.scala]: Oh, I didn't realize I could do that. I simplified it even more: callback.foreach(_ (Left(new HeavyLoadException)))

**Comment by vikstrous** [comment]
I don't think I know enough about Norbert to add more useful tests and I feel like all the testing I can really do is just duplicating existing tests for sendRequest. Any ideas about anything specific to test?

**Comment by jhartman** [comment]
Tests look good to me. I'll merge in.

**Pull request merged**